pkg/api/pluginproxy/token_cache.go

```diff
@@ -0,0 +1,124 @@
+package pluginproxy
+
+import (
+	"context"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
```

Comment:
This cache is pretty generic, but with this import it kind of becomes an azure cache. Maybe rename it to azure_token_cache.go?

Reply:
Yes, I wanted to make it generic.
I'm not sure whether it makes sense to refactor other token providers to use it too, since the Azure token provider eventually will go away into Azure library and this cache will probably follow as well.
I have some further optimizations in mind they depend whether this token cache becomes Azure specific or not.
I removed dependency on `azcore` but I will postpone any further changes to it until we decided it's fate.